### PR TITLE
[Improve][dist]add shade check rule

### PR DESCRIPTION
**Pull request by hawk9821**
### Purpose of this pull request
Globally check that shade related classes are used correctly in project.   Log format：
```
org.opentest4j.AssertionFailedError: guava shade is not up to code, need add prefix [org.apache.seatunnel.shade.]. 
E:\work\seatunnel\seatunnel-api\src\main\java\org\apache\seatunnel\api\table\type\MapType.java
com.google.common.base.Preconditions.checkArgument  [27]
com.google.common.base.Preconditions.checkNotNull  [28]
E:\work\seatunnel\seatunnel-core\seatunnel-spark-starter\seatunnel-spark-starter-common\src\main\java\org\apache\seatunnel\core\starter\spark\execution\SourceExecuteProcessor.java
com.google.common.collect.Lists  [20]
 ==> 
Expected :0
Actual   :2
<Click to see difference>
```

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-shade/seatunnel-guava/pom.xml [hunk 1]
```diff
@@ -62,8 +62,8 @@
                             </filters>
                             <relocations>
                                 <relocation>
-                                    <pattern>com.google</pattern>
-                                    <shadedPattern>${seatunnel.shade.package}.com.google</shadedPattern>
+                                    <pattern>com.google.common</pattern>
```
Comment (Hisoka-X) [inline seatunnel-shade/seatunnel-guava/pom.xml]: why only shade common prefix?

File: seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                            errorMap.put(clazzPath, collect);
+                        }
+                    }
+                });
+        return errorMap;
+    }
+
+    private String errorMsg(String checkType, Map<String, List<String>> errorMap) {
+        StringBuilder msg = new StringBuilder();
+        msg.append(String.format("%s shade is not up to code, need add prefix [", checkType))
```
Comment (Hisoka-X) [inline seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java]: can we tell to devs the code file name?

File: seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+@Slf4j
+public class ImportShadeClassCheck {
+
+    private static Map<String, NodeList<ImportDeclaration>> importsMap = new HashMap<>();
+    private final String SEATUNNEL_SHADE_PREFIX = "org.apache.seatunnel.shade.";
+    private final boolean isWindows = System.getProperty("os.name").toLowerCase().startsWith("win");
+
+    @BeforeAll
+    public static void beforeAll() {
+        Path directoryPath = Paths.get(System.getProperty("user.dir"));
```
Comment (Hisoka-X) [inline seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java]: Does all seatunnel project contains this property?

File: seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java (hunk 2 above)
Comment (hawk9821) [reply seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java]: log format: 
```
org.opentest4j.AssertionFailedError: guava shade is not up to code, need add prefix [org.apache.seatunnel.shade.]. 
E:\work\seatunnel\seatunnel-api\src\main\java\org\apache\seatunnel\api\table\type\MapType.java
com.google.common.base.Preconditions.checkArgument  [27]
com.google.common.base.Preconditions.checkNotNull  [28]
E:\work\seatunnel\seatunnel-core\seatunnel-spark-starter\seatunnel-spark-starter-common\src\main\java\org\apache\seatunnel\core\starter\spark\execution\SourceExecuteProcessor.java
com.google.common.collect.Lists  [20]
 ==> 
Expected :0
Actual   :2
<Click to see difference>
```

File: seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java (hunk 3 above)
Comment (hawk9821) [reply seatunnel-dist/src/test/java/org/apache/seatunnel/ImportShadeClassCheck.java]:
```
2024-11-27T03:20:25.0504611Z [INFO] Running org.apache.seatunnel.api.ImportShadeClassCheckTest
2024-11-27T03:20:25.0505865Z [] 2024-11-27 03:20:25,024 INFO  org.apache.seatunnel.api.ImportShadeClassCheckTest - work directory ===>  D:\a\seatunnel\seatunnel\seatunnel-dist
2024-11-27T03:20:38.9589159Z [] 2024-11-27 03:20:38,920 INFO  org.apache.seatunnel.api.ImportShadeClassCheckTest - check jackson shade successfully
2024-11-27T03:20:39.0284906Z [] 2024-11-27 03:20:39,014 INFO  org.apache.seatunnel.api.ImportShadeClassCheckTest - check guava shade successfully
2024-11-27T03:20:39.1050318Z [] 2024-11-27 03:20:39,092 INFO  org.apache.seatunnel.api.ImportShadeClassCheckTest - check janino shade successfully
2024-11-27T03:20:39.1603157Z [] 2024-11-27 03:20:39,155 INFO  org.apache.seatunnel.api.ImportShadeClassCheckTest - check jetty shade successfully
2024-11-27T03:20:39.1683114Z [INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.115 s - in org.apache.seatunnel.api.ImportShadeClassCheckTest
```
we can see from the log that the directory for `System.getProperty("user.dir") ` is  `D:\a\seatunnel\seatunnel\seatunnel-dist` ,  the scan directory is `D:\a\seatunnel\seatunnel\`

**Review by Hisoka-X — DISMISSED** [review]

**Comment by Hisoka-X** [comment]
Please resolve conflicts.

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**